.github/workflows/integration-pr.yml

```diff
@@ -0,0 +1,198 @@
+# **This is not a trusted workflow.**
+# **what?** - This workflow runs integration tests for supported OS,
+# python versions and core adapters. It will only run tests for
+# related file changes. Use the `test all` and `test ${adapter}`
+# label to manually run all or additional tests. Use `ok to test`
+# label to mark PRs from forked repositories that are safe to run integration
+# tests for. Requires secrets to run against different warehouses.
+# **why?** - This checks the functionality of dbt from a user's perspective,
+# provides code coverage, and attempts to catch functional regressions.
```

Comment:
For the code coverage aspect, does it just print the results to the console? Can we have it output as an artifact?

Reply:
Oh, I thought I'd be able to set up coverage reporting, but it is trickier than I thought! I'll remove that statement for now.

Here is the coverage tool that integrates with pytest that I was trying to use: https://pytest-cov.readthedocs.io/en/latest/reporting.html

I actually like how codecov works, you can send many coverage reports for the same PR run and it will combine them on server side and we are able to set up checks that _new_ code is has test coverage. This is awesome because it will work great with unit tests and integration tests in different workflows and it is possible only a subset of integration tests could run for a PR. I like this approach vs looking at project wide code coverage! (https://github.com/codecov/codecov-action)

This is also a pretty cool idea for reporting tests failures in the codebase, though, it could get noisy: https://github.com/utgwkk/pytest-github-actions-annotate-failures